src/driver/franka_plan_runner.cc

```diff
@@ -171,159 +213,146 @@ franka::RobotMode FrankaPlanRunner::GetRobotMode(franka::Robot& robot) {
-            if (status_ == RobotStatus::Running) {
-              dexai::log()->info(
-                  "RunFranka: Robot is running and waiting for a new plan...");
-            } else if (status_ == RobotStatus::Paused) {
-              dexai::log()->info(
-                  "RunFranka: Robot is paused, waiting to get unpaused...");
-            } else {
-              dexai::log()->error(
-                  "RunFranka: Robot is {}, this state should not have happened",
-                  utils::RobotStatusToString(status_));
-            }
-            status_has_changed = false;  // reset
-            t_last_main_loop_log_ = t_now;
-          }
-          // only publish robot_status, do that twice as fast as the lcm publish
-          // rate ...
-          // TODO: add a timer to be closer to lcm_publish_rate_ [Hz] * 2.
-          robot.read([this](const franka::RobotState& robot_state) {
-            auto cannonical_robot_state =
-                ConvertToCannonical(robot_state, joint_pos_offset_);
-            // publishing cannonical values over lcm
-            comm_interface_->SetRobotData(cannonical_robot_state,
-                                          next_conf_plan_);
-            std::this_thread::sleep_for(std::chrono::milliseconds(
-                static_cast<int>(1000.0 / (lcm_publish_rate_ * 2.0))));
-            return false;
-          });
-        }
       } catch (const franka::ControlException& ce) {
         status_has_changed = true;
-        dexai::log()->warn("RunFranka: Caught control exception: {}.",
+        dexai::log()->warn("RunFranka: control exception in main loop: {}.",
                            ce.what());
-
-        if (!RecoverFromControlException(robot)) {
-          dexai::log()->error(
-              "RunFranka: RecoverFromControlException did not work!");
-          comm_interface_->PublishDriverStatus(false, ce.what());
+        comm_interface_->PublishDriverStatus(false, ce.what());
+        if (!RecoverFromControlException(*robot_)) {
+          dexai::log()->error("RunFranka: RecoverFromControlException failed");
           return 1;
         }
       }
+      continue;
     }
-
-  } catch (const franka::Exception& ex) {
-    dexai::log()->error(
-        "RunFranka: Caught expection during initilization, msg: {}", ex.what());
-    comm_interface_->PublishDriverStatus(false, ex.what());
-    return 1;  // bad things happened.
+    // no new plan available in the buffer or robot isn't running
+    if (comm_interface_->CancelPlanRequested()) {
+      log()->error(
+          "RunFranka: plan cancellation requested but there is no active "
+          "plan");
+      comm_interface_->ClearCancelPlanRequest();
+    }
+    // print out status if there's any change, or every 10 sec
+    if (auto t_now {std::chrono::steady_clock::now()};
+        status_has_changed
+        || t_now - t_last_main_loop_log_ >= std::chrono::seconds(10)) {
```

Comment:
How should we specify these wait times? We have a 1-second wait earlier and a 10-second wait here.

Reply:
they are different things, this is only for the status printout